### PR TITLE
Setup changes and static analysis

### PR DESCRIPTION
**Pull request by caffalaughrey**

### CODE REVIEW

**Comment by caffalaughrey** [comment]
@demianbrecht thanks for all the commits.  I've moved them to this PR as right now feature branch commits are going through `forcedotcom:developer` first.  I'll close the old PR's, but would you mind hanging on to the branches for now?

I should have this reviewed for you for next Thursday.  In the mean time, looks like the build is failing after merging with earlier commits from `forcedotcom:developer`.  I resolved a few conflicts in `setup.py`, but I didn't get a chance to look at this extensively.  Can you take a look if you have the chance?

**Comment by demianbrecht** [comment]
Would it perhaps make sense to have `developer` branch be default then? I believe that PRs would default to that instead of `master`. Otherwise you'll likely have to deal with this every time a PR is submitted ;)

Yeah I'll take a look at the failures. I'll try to do so early next week at the latest.

**Comment by caffalaughrey** [comment]
Good point; default changed.

File: setup.cfg [hunk 1]
```diff
@@ -1,2 +1,9 @@
...
-test=pytest
\ No newline at end of file
+test=pytest
+
+[tool:pytest]
+addopts=--flake8
+
+[flake8]
+max-line-length=120
+ignore=W605
```
Comment (ccheevers) [inline setup.cfg]: I see that flake 8 is ignoring invalid escape sequences (https://lintlyci.github.io/Flake8Rules/rules/W605.html), did a build fail due to this? @demianbrecht

**Pull request merged**